Automattic/jetpack
# Improve Google Verification Service integration

Fixes #10141
Requires server patch D18402-code

#### Changes proposed in this Pull Request:

* Do not verify the site on page load, even if the user already has a google site verification token, wait for the user to click on the "Auto verify with Google" button
* Display an error notice on request error
* Revamp of the google site verification UI, props to @jeffgolenski for his design proposal!

#### Testing instructions:
- Go to `/wp-admin/admin.php?page=jetpack#/traffic` (Jetpack > Settings > Traffic) and scroll down to Site Verification
- Try clicking on "Auto verify with Google", authenticate with Google and make sure your site is verified, go to https://www.google.com/webmasters/verification/home?hl=en&authuser=2 and make sure your site is in the list. Go to https://search.google.com/search-console and make sure your site is there as well
- You should see a link to the Google Search Console, click on it and verify that you can access your site there. Go to Sitemaps and check that your sitemap is added there (it will probably fail for `wpsandbox.me` sites as `/robots.txt` prevents googlebot from crawling the site)
- Unverify your site by clicking on the Manually verify with Google, emptying the field and saving, then go to https://www.google.com/webmasters/verification/home?hl=en&authuser=2 click on "Verification Details" and unverify for your email. You can also go to https://search.google.com/search-console select the property and remove it after the verification fails

#### Proposed changelog entry for your changes:
- Improve Google Verification Service integration
